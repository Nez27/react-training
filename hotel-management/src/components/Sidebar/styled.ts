import styled from 'styled-components';

const StyledSidebar = styled.aside`
  padding: 50px 20px;

  grid-row: 1 / 3;

  border: 1px solid var(--border-color);
`;

const Heading = styled.h1`
  font-size: var(--fs-md);
  text-transform: uppercase;
  text-align: center;

  cursor: pointer;

  color: var(--primary-color);
`;

export { StyledSidebar, Heading };
