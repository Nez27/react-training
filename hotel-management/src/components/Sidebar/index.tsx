// Components
import Nav from '../Nav';

// Styled
import { Heading, StyledSidebar } from './styled';

interface ISidebar {
  heading: string;
}

const Sidebar = ({ heading }: ISidebar) => {
  return (
    <StyledSidebar>
      <Heading>{heading}</Heading>
      <Nav />
    </StyledSidebar>
  );
};

export default Sidebar;
