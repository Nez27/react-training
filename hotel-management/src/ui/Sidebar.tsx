import styled from 'styled-components';

// Components
import Nav from './Nav';

const StyledSidebar = styled.aside`
  padding: 50px 20px;

  grid-row: 1 / 3;
  border: 1px solid var(--border-color);
`;

const Heading = styled.h1`
  font-size: 30px;
  text-transform: uppercase;
  text-align: center;
  color: var(--primary-color);
`;

const Sidebar = () => {
  return (
    <StyledSidebar>
      <Heading>Hotel Management</Heading>
      <Nav />
    </StyledSidebar>
  );
};

export default Sidebar;
