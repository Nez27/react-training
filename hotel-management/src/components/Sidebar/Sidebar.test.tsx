import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

// Components
import Sidebar from '.';

describe('Sidebar', () => {
  const wrapper = renderer.create(
    <BrowserRouter>
      <Sidebar heading="Hotel Management" />
    </BrowserRouter>
  );

  test('Should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
