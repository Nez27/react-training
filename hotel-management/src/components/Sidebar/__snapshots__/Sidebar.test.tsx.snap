// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Should render correctly 1`] = `
<aside
  className="sc-dlniIP dfcuiS"
>
  <h1
    className="sc-hKFymg OpLao"
    onClick={[Function]}
  >
    Hotel Management
  </h1>
  <nav
    className="sc-bdnyFh gGCifp"
  >
    <a
      className="sc-gtsqUy gbSQWi"
      href="/booking"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5 22h14c1.103 0 2-.897 2-2V6c0-1.103-.897-2-2-2h-2V2h-2v2H9V2H7v2H5c-1.103 0-2 .897-2 2v14c0 1.103.897 2 2 2zM5 7h14v2H5V7z"
        />
      </svg>
      <span>
        Booking
      </span>
    </a>
    <a
      className="sc-gtsqUy gbSQWi"
      href="/user"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4.5 6.375a4.125 4.125 0 118.25 0 4.125 4.125 0 01-8.25 0zM14.25 8.625a3.375 3.375 0 116.75 0 3.375 3.375 0 01-6.75 0zM1.5 19.125a7.125 7.125 0 0114.25 0v.003l-.001.119a.75.75 0 01-.363.63 13.067 13.067 0 01-6.761 1.873c-2.472 0-4.786-.684-6.76-1.873a.75.75 0 01-.364-.63l-.001-.122zM17.25 19.128l-.001.144a2.25 2.25 0 01-.233.96 10.088 10.088 0 005.06-1.01.75.75 0 00.42-.643 4.875 4.875 0 00-6.957-4.611 8.586 8.586 0 011.71 5.157v.003z"
        />
      </svg>
      <span>
        User
      </span>
    </a>
    <a
      className="sc-gtsqUy gbSQWi"
      href="/room"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6.5 14.5v-3.505c0-.245.25-.495.5-.495h2c.25 0 .5.25.5.5v3.5a.5.5 0 0 0 .5.5h4a.5.5 0 0 0 .5-.5v-7a.5.5 0 0 0-.146-.354L13 5.793V2.5a.5.5 0 0 0-.5-.5h-1a.5.5 0 0 0-.5.5v1.293L8.354 1.146a.5.5 0 0 0-.708 0l-6 6A.5.5 0 0 0 1.5 7.5v7a.5.5 0 0 0 .5.5h4a.5.5 0 0 0 .5-.5Z"
        />
      </svg>
      <span>
        Room
      </span>
    </a>
  </nav>
</aside>
`;
